blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DAbstractMetricsRW.java

```
@@ -0,0 +1,324 @@
+     */
+    @Override
+    public void insertRollups(List<SingleRollupWriteContext> writeContexts) throws IOException {
+
+        if (writeContexts.size() == 0) {
+            return;
+        }
+
+        Map<ResultSetFuture, SingleRollupWriteContext> futureLocatorMap = new HashMap<ResultSetFuture, SingleRollupWriteContext>();
+        Timer.Context ctx = Instrumentation.getWriteTimerContext( writeContexts.get( 0 ).getDestinationCF().getName() );
+        try {
+            for (SingleRollupWriteContext writeContext : writeContexts) {
+                Rollup rollup = writeContext.getRollup();
+                Locator locator = writeContext.getLocator();
+                Granularity granularity = writeContext.getGranularity();
+                int ttl = getTtl(locator, rollup.getRollupType(), granularity);
+
+                // lookup the right writer
+                RollupType rollupType = writeContext.getRollup().getRollupType();
+                DAbstractMetricIO io = rollupTypeToIO.get(rollupType);
+                if ( io == null ) {
+                    throw new InvalidDataException(
+                            String.format("insertRollups(locator=%s, granularity=%s): unsupported rollupType=%s",
+                                    locator, granularity, rollupType.name()));
+                }
+
+                ResultSetFuture future = io.putAsync(locator, writeContext.getTimestamp(), rollup, writeContext.getGranularity(), ttl);
+                futureLocatorMap.put(future, writeContext);
+            }
+
+            for (ResultSetFuture future : futureLocatorMap.keySet()) {
+                try {
+                    future.getUninterruptibly().all();
+                } catch (Exception ex) {
+                    SingleRollupWriteContext writeContext = futureLocatorMap.get(future);
+                    LOG.error(String.format("error writing to locator %s, granularity %s", writeContext.getLocator(), writeContext.getGranularity()), ex);
```

We are just eating exception here.